plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -238,32 +238,46 @@ public static Table getIcebergTableWithMetadata(
 
     public static List<IcebergColumnHandle> getProjectedColumns(Schema schema, TypeManager typeManager)
     {
-        ImmutableList.Builder<IcebergColumnHandle> projectedColumns = ImmutableList.builder();
-        StructType schemaAsStruct = schema.asStruct();
-        Map<Integer, NestedField> indexById = TypeUtil.indexById(schemaAsStruct);
-        Map<Integer, Integer> indexParents = TypeUtil.indexParents(schemaAsStruct);
+        Map<Integer, NestedField> indexById = TypeUtil.indexById(schema.asStruct());
+        return getSelectedProjectColumns(schema, typeManager, indexById, indexById.keySet() /* project all columns */);
+    }
+
+    public static List<IcebergColumnHandle> getSelectedProjectColumns(Schema schema, TypeManager typeManager, Set<Integer> fieldIds)
```

Comment:
I think the previous name of `getProjectedColumns` was fine, current name is a bit confusing